# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxMacPlatformFontList.mm
# Commit: 277876c31869
# Full Hash: 277876c31869cad23a2e93aab7bb4c7ee0c5f0f9
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-08-13 09:54:32
# Regressor Bug: 1565966
# File Overlap Count: 1
# Description:
#   Bug 1565966 - part 1 - Normalize font family names to lowercase in mAliasTable, for consistency with eventual font-family lookups. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D41242
# ==============================================================================

diff -r ed6b960586ea -r 277876c31869 gfx/thebes/gfxMacPlatformFontList.mm
--- a/gfx/thebes/gfxMacPlatformFontList.mm	Mon Aug 12 18:23:28 2019 +0000
+++ b/gfx/thebes/gfxMacPlatformFontList.mm	Mon Aug 12 17:49:09 2019 +0000
@@ -1065,13 +1065,13 @@
       if (face->mDescriptor.AsString(list).Equals(familyName)) {
         // Found it! Create an entry in the Alias table.
         GenerateFontListKey(familyName, key);
-        if (SharedFontList()->FindFamily(key) || mAliasTable.Get(familyName)) {
+        if (SharedFontList()->FindFamily(key) || mAliasTable.Get(key)) {
           // If the family name is already known, something's misconfigured;
           // just ignore it.
           MOZ_ASSERT(false, "single-face family already known");
           break;
         }
-        auto af = mAliasTable.LookupOrAdd(familyName);
+        auto af = mAliasTable.LookupOrAdd(key);
         af->AppendElement(facePtrs[i]);
         break;
       }
@@ -1792,7 +1792,9 @@
     gfxFontUtils::ReadOtherFamilyNamesForFace(canonicalName, nameData, dataLength, otherFamilyNames,
                                               false);
     for (const auto& alias : otherFamilyNames) {
-      auto af = mAliasTable.LookupOrAdd(alias);
+      nsAutoCString key;
+      GenerateFontListKey(alias, key);
+      auto af = mAliasTable.LookupOrAdd(key);
       af->AppendElement(facePtrs[i]);
     }
   }
